src/lib.rs

```diff
@@ -765,6 +772,49 @@ impl CertificateParams {
 						if !self.subject_alt_names.is_empty() {
 							self.write_subject_alt_names(writer.next());
 						}
+
+						// Write standard key usage
+						if !self.key_usages.is_empty() {
+							writer.next().write_sequence(|writer| {
+
+								let oid = ObjectIdentifier::from_slice(OID_KEY_USAGE);
+								writer.next().write_oid(&oid);
+								writer.next().write_bool(true);
+
+								let mut bits = BitVec::new();
+
+
+								for entry in self.key_usages.iter() {
+
+									// Map the index to a value
+									let index = match entry {
+										KeyUsagePurpose::DigitalSignature => 0,
+										KeyUsagePurpose::NonRepudiation => 1,
+										KeyUsagePurpose::KeyEncipherment => 2,
+										KeyUsagePurpose::DataEncipherment => 3,
+										KeyUsagePurpose::KeyAgreement => 4,
+										KeyUsagePurpose::KeyCertSign => 5,
+										KeyUsagePurpose::CrlSign => 6,
+										KeyUsagePurpose::EncipherOnly => 7,
+										KeyUsagePurpose::DecipherOnly => 8,
+									};
+
+									// Expand the number of bits
+									if bits.len() < (index + 1) {
+										bits.grow( (index + 1) - bits.len(), false);
+									}
+
+									bits.set(index, true);
+								}
+
+								let der = yasna::encode_der(&bits);
+
+								// Write them
+								writer.next().write_bytes(&der);
```

Comment:
Before we're adding an entire dependency, what about trying `write_bitvec_bytes` instead? It shouldn't be too complicated to manually write a bitvec. It seems it can't be more than 9 bits anyways, no?

Reply:
Logic or'ing some bits is easy. Getting the format correctly has been proving to be troublesome. I don't have another solution for this unfortunately.

Reply:
Gave this another whirl. I think I got it. Force pushed the changes.